kairon/shared/channels/broadcast/whatsapp.py

```
@@ -495,24 +522,18 @@ def __get_client(self):
-            access_token = channel_config["config"].get('api_key') or channel_config["config"].get('access_token')
+            bsp_type = channel_config["config"].get("bsp_type")
+            if bsp_type == WhatsappBSPTypes.bsp_gupshup.value:
+                access_token = channel_config["config"].get("partner_app_token")
+            else:
+                access_token = channel_config["config"].get('api_key') or channel_config["config"].get('access_token') or channel_config["config"].get("partner_app_token")
             channel_client = WhatsappFactory.get_client(bot_settings["whatsapp"])
             channel_client = channel_client(access_token, config=channel_config)
             return channel_client
         except DoesNotExist as e:
             logger.exception(e)
             raise AppException(f"Whatsapp channel config not found!")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)